### ISSUE TITLE
document deployment infrastructure

### ISSUE THREAD

**Issue by tkphd** [hpc-carpentry/hpc-carpentry.github.io#20]
Strange that `build-and-deploy.yml` uses `gh-pages`, not the default `main` branch where edits go. 

https://github.com/hpc-carpentry/hpc-carpentry.github.io/blob/a1ea1ddc03449af745a790c6af787538e55293d9/.github/workflows/build-and-deploy.yml#L53

Apparently, this is correct (see below), and necessary because this is not a vanilla deployment (i.e., not hosted by GitHub).

**Comment by ocaisa** [comment]
Yes, this action allows us to do the complicated build processes that are not allowed by vanilla GitHub pages, the action takes the main branch, builds it and then deploys it  to `gh-pages`

**Comment by tkphd** [comment]
Okaaay... what "complicated build processes" are we up to, and are they necessary?

**Comment by ocaisa** [comment]
You need to talk to me about 6 months ago when this information was on the tip of my tongue

**Comment by reid-a** [comment]
Amend this issue to "document this workflow"?

**Comment by ocaisa** [comment]
I think it has something to do with queries to the GitHub repos and the use of json files

**Comment by ocaisa** [comment]
Carpentries should know, they do not deploy via GitHub because of these requirements

**Comment by ocaisa** [comment]
It's because of `jekyll-get-json`, see https://github.com/hpc-carpentry/hpc-carpentry.github.io/commit/a3a475f80701e71600be8bdc1d5b8568935480a8 for a little context
